### PR TITLE
Allow domains admins to edit SMS Gateway config

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Summary

https://dimagi-dev.atlassian.net/browse/SAAS-11912
## Product Description

This will allow domain admins to view/edit SMS gateway which was resulting in 404 before
## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story

This has been tested locally and I was able to access the page from domain admin accounts and superuser but not from other users.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/sms/views.py [hunk 1]
```diff
@@ -1117,9 +1121,13 @@ def hq_api_id(self):
...
     def backend_class(self):
-        # Superusers can create/edit any backend
+        # Superusers and domain admins can create/edit any backend
         # Regular users can only create/edit Telerivet backends for now
-        if not self.is_system_admin and self.hq_api_id != SQLTelerivetBackend.get_api_id():
+        if (
+            not self.is_system_admin
+            and not self.is_domain_admin
+            and self.hq_api_id != SQLTelerivetBackend.get_api_id()
+        ):
```
Comment (mkangia) [inline corehq/apps/sms/views.py:1130]: is this same as?
```
if not (
  self.is_system_admin
  or self.is_domain_admin
  or self.hq_api_id != SQLTelerivetBackend.get_api_id()
)
```
If yes,
that is easier to relate to with the comment.

Is there a need of a sign off on this change from product? I am not sure. Just thinking if it was kept that way for a reason.

Comment (millerdev) [reply corehq/apps/sms/views.py:1130]: Think it should be
```py
if not (
    self.is_system_admin
    or self.is_domain_admin
    or self.hq_api_id == SQLTelerivetBackend.get_api_id()
)
```

Comment (AmitPhulera) [reply corehq/apps/sms/views.py:1130]: Yep. That is much clearer to read. Thanks, @millerdev @mkangia for your inputs. I see that there was an explicit test that was added to ensure that domain admins do not edit Twilio Backend which is failing in this case. 
I will ask Product for the inputs on the tickets and proceed accordingly.

**Comment by AmitPhulera** [comment]
Confirmed from the product that we don't want domain admins to edit the SMS Gateway Configuration. So I am closing this one out.

**Pull request closed without merging**